pkg/config/dynamicloader_test.go

```diff
@@ -0,0 +1,612 @@
+//nolint:golint,goconst
```

Comment:
Ditto with my old comment about this file being quite long and maybe having an opportunity to make the set of assertions smaller by finding another way to make comparisons

Reply:
Thoughts on splitting the tests between config loading and template? I also removed some redundant tests. Splitting/removing redundant tests means one file is 175 lines and the other is 350 lines

Reply:
Seems reasonable 👍